minetest/minetest
# Allow nodes to have their post_effect_color affected by lighting

Closes #12071.

This PR adds a new boolean field called `post_effect_color_shaded` to node definitions. If this field is true, the `post_effect_color` of the node will be affected by lighting. The default value is false, so nothing changes for existing mods.
Applying lighting to `post_effect_color` makes sense because `post_effect_color` is not applied by color multiplication, but by alpha blending. Lighting is done the same way as for the wielditem.

Here are two videos so you can see the effect:
`post_effect_color_shaded = false`: https://share.grecloud.de/without%20post_effect_color_shaded%202.mp4
`post_effect_color_shaded = true`: https://share.grecloud.de/with%20post_effect_color_shaded%202.mp4

The videos were taken with Minetest Game, the Soothing 32 texture pack and `post_effect_color = {r = 40, g = 204, b = 223, a = 100}` for `default:water_source` and `default:water_flowing`.
Notice that in the first video, the scene is much brighter at night when I'm underwater than when I'm not. In contrast, in the second video, the brightness remains relatively constant at night, regardless of whether I'm underwater or not. To me, the second video looks much better.

**Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?**
Maybe "2.1 Rendering/Graphics improvements".

## To do

This PR is a Ready for Review.

## How to test

Verify that the `post_effect_color` of existing nodes still looks the same.

Try adding `post_effect_color_shaded = true` to the definition of an existing node. Verify that the `post_effect_color` of the node is now affected by lighting. This isn't only visible at night, but also when you go into deep, dark water.
